### ISSUE TITLE
Assertion failure on invalid query syntax

### ISSUE THREAD

**Issue by alemuller** [tree-sitter/tree-sitter#968]
Assertion failure on missing closing bracket. Expecting invalid syntax error message.

MWE with tree-sitter-c.

`tree-sitter-c/queries/highlights.scm`:
```scm highlights.scm
(parameter_list
  [ ")" @foo)
```

```
$ tree-sitter --version
tree-sitter 0.19.1 (e20aff9a9c5be212018b7de1247fba98215d85b2)
$ touch repro.c
$ tree-sitter highlight repro.c
tree-sitter: src/./query.c:1200: ts_query__analyze_patterns: Assertion `exists' failed.
```

**Comment by theHamsta** [comment]
![image](https://user-images.githubusercontent.com/7189118/110225564-f866da00-7ee6-11eb-977c-438f4683bd91.png)

Also happens with our ocaml highlights: https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/ocaml/highlights.scm

**Comment by maxbrunsfeld** [comment]
@theHamsta Can you reduce that error at all? Is there a minimal query that triggers the error with the Ocaml grammar?

**Comment by theHamsta** [comment]
This seems to be the problematic part

```
(attribute ["[@" "]"] @punctuation.special)
```

**Comment by maxbrunsfeld** [comment]
Awesome, thanks for isolating that. Should be a pretty quick fix now.

**Comment by maxbrunsfeld** [comment]
Ok, that is also fixed in #971. It was indeed a somewhat related problem.

**Comment by theHamsta** [comment]
Thanks! Can confirm that it works.
